components/ILIAS/Chatroom/classes/class.BannedUsersTable.php

```diff
@@ -0,0 +1,212 @@
+
+            'actor' => $this->ui_factory->table()->column()->text($this->lng->txt('chtr_ban_actor_tbl_head'))
+                                        ->withIsSortable(true),
+        ];
+    }
+
+    protected function getActions(): array
+    {
+        $query_params_namespace = ['chat_ban_table'];
+
+        $uri_detach = $this->data_factory->uri(
+            ILIAS_HTTP_PATH . '/' . $this->ctrl->getLinkTargetByClass(ilObjChatroomgui::class, 'ban-delete')
+        );
+
+        $url_builder_detach = new UI\URLBuilder($uri_detach);
+        list(
+            $url_builder_detach, $action_parameter_token_copy, $row_id_token_detach
+            ) =
+            $url_builder_detach->acquireParameters(
+                $query_params_namespace,
+                'action',
+                'user_ids'
+            );
+
+        return [
+            'ban-delete' => $this->ui_factory->table()->action()->multi(
+                $this->lng->txt('unban'),
+                $url_builder_detach->withParameter($action_parameter_token_copy, 'ban-delete'),
+                $row_id_token_detach
+            ),
+        ];
+    }
+
+    protected function getDataRetrieval(): UI\Component\Table\DataRetrieval
+    {
+        $data_retrieval = new class($this->banned_users) implements UI\Component\Table\DataRetrieval {
+
+            /**
+             * @var array|string[]
+             */
+            private array $numericColumns;
+
+            private ?array $records = null;
+
+            public function __construct(protected array $banned_users)
+            {
+                $this->numericColumns = ['user_id'];
+            }
+
+            private function initRecords(): void
+            {
+                if ($this->records === null) {
+                    $this->records = [];
+                    $i = 0;
+                    $entries = $this->banned_users;
+
+                    foreach ($entries as $entry) {
+                        $user_id = $entry['user_id'];
+                        /** @var ilObjUser $user */
+                        $user = ilObjectFactory::getInstanceByObjId($user_id, false);
+                        if (!($user instanceof ilObjUser)) {
+                            continue;
+                        }
+
+                        $this->records[$i]['user_id'] = $user->getId();
+                        $this->records[$i]['login'] = $user->getLogin();
+                        $this->records[$i]['firstname'] = $user->getFirstname();
+                        $this->records[$i]['lastname'] = $user->getLastname();
+                        if (is_numeric($entry['timestamp']) && $entry['timestamp'] > 0) {
+                            $this->records[$i]['timestamp'] = ilDatePresentation::formatDate(
+                                new ilDateTime($entry['timestamp'], IL_CAL_UNIX)
+                            );
+                        }
+
+                        $this->records[$i]['actor'] = $entry['actor'];
```

Comment:
Do we really need to create a new user instance here? IMO all the data should be available when calling `\ilChatroom::getBannedUsers`.